### ISSUE TITLE
New Allocator Fails On >=4GB Requests

### ISSUE THREAD

**Issue by c-blake** [nim-lang/Nim#7120]
```Nim
proc myNewSeq*(capacity: uint64): seq[uint64] =
  result = newSeq[uint64](capacity)
#discard myNewSeq((1 shl 29) - 9)  # works
discard myNewSeq((1 shl 29) - 8)  # writes "out of memory"
#discard myNewSeq((1 shl 29) - 4)  # even worse rawNewObj ends up nil and zeroMem aborts
```
Personally, I allocate >4GB single seqs all the time.  Even my phone has that much RAM.

I believe I tracked down the error to 32-bit arithmetic in alloc.nim:mappingSearch.  { I just did a system call trace and noticed the roughly 4GB request translated to an mmap of 2**58 bytes.  I then set a breakpoint at that crazy mmap and saw getBigChunk was the last call with a reasonable size request.  Then reading the code for getBigChunk implicated mappingSearch. }

It looks like recent (Dec 7) code and that some 32-bit limitations are documented.  So, this code failing on large request sizes may have been something that you are already vaguely aware of, but it seemed worth opening this issue.

**Comment by Araq** [comment]
TBH I didn't imagine anybody to use allocations bigger than 2GB. ;-)

**Comment by c-blake** [comment]
3 years ago I had access to a box with 1TB RAM and another 1TB swap space..room for literally 100s of seqs that might exceed 4GB all at the same time. ;-)

Anyway, https://github.com/mattconte/tlsf/ makes it seem like the changes to support 64-bits are pretty simple.  (Just search for the code conditioned on TLSF_64BIT).  That original TLSF paper was written in 2004 when 64-bit pointers were much more rare.  For Nim it might be best to just use int64/uint64 unconditionally.

**Comment by c-blake** [comment]
Sure.  I should be able to work on this sometime over the next week.  Will send a PR when I have something.

**Comment by timotheecour** [comment]
FYI:
tfragment_alloc.nim (which allocates 4GB) often makes appveyor fail with out of memory #8509
fix #8509 disable 4GB allocating test on windows that crashed appveyor #8510
